app/controlplane/pkg/biz/casmapping.go

```diff
@@ -129,24 +148,19 @@ func (uc *CASMappingUseCase) FindCASMappingForDownloadByOrg(ctx context.Context,
 	if err != nil {
 		return nil, fmt.Errorf("failed to load mappings associated to an user: %w", err)
 	} else if len(orgMappings) > 0 {
-		result := defaultOrFirst(orgMappings)
-
-		uc.logger.Infow("msg", "mapping found!", "digest", digest, "orgs", orgs, "casBackend", result.CASBackend.ID, "default", result.CASBackend.Default, "public", result.Public)
-		return result, nil
+		return defaultOrFirst(orgMappings), nil
 	}
 
 	// 3 - mappings that are public
 	publicMappings := filterByPublic(mappings)
 	// The user has not access to neither proprietary nor public mappings
 	if len(publicMappings) == 0 {
 		uc.logger.Warnw("msg", "digest exist but user does not have access to it", "digest", digest, "orgs", orgs)
-		return nil, NewErrUnauthorized(errors.New("unauthorized access to the artifact"))
+		return nil, NewErrNotFound("digest not found in any mapping")
```

Comment:
I can't see the reason we wanted to differentiate this from NotFound in practice.